test/e2e/storage/vsphere_volume_cluster_ds.go

```diff
@@ -0,0 +1,121 @@
+package storage
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere/vclib"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+/*
+	Tests to verify volume provisioning on a clustered datastore
+	1. Static provisioning
+	2. Dynamic provisioning
+	3. Dynamic provisioning with spbm policy
+
+	This test reads env
+	1. CLUSTER_DATASTORE which should be set to clustered datastore
+	2. VSPHERE_SPBM_POLICY_DS_CLUSTER which should be set to a tag based spbm policy tagged to a clustered datastore
+*/
+var _ = SIGDescribe("Volume Provisioning On Clustered Datastore [Feature:vsphere]", func() {
+	f := framework.NewDefaultFramework("volume-provision")
+
+	var client clientset.Interface
+	var namespace string
+	var scParameters map[string]string
+	var clusterDatastore string
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		scParameters = make(map[string]string)
+
+		clusterDatastore = os.Getenv("CLUSTER_DATASTORE")
+		Expect(clusterDatastore).NotTo(BeEmpty(), "Please set CLUSTER_DATASTORE system environment")
```

Comment:
perhaps this error msg can show an example value of a valid CLUSTER_DATASTORE value?

Reply:
Adding a format example for a clustered datastore name. Example with a specific value
like `dscl1/sharedVmfs-0` wouldn't be much helpful here IMO.